### ISSUE TITLE
rpc:call try/catch rex

### ISSUE THREAD

**Issue by joedevivo** [basho/riak_core#579]
```
/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core.erl:
   94  
   95  get_other_ring(Node) ->
   96:     case rpc:call(Node, riak_core_ring_manager, get_my_ring, []) of
   97          {ok, Ring} ->
   98              case riak_core_ring:legacy_ring(Ring) of
   ..
  100                      {ok, Ring};
  101                  false ->
  102:                     rpc:call(Node, riak_core_ring_manager, get_raw_ring, [])
  103              end;
  104          Error ->
  ...
  160      case net_adm:ping(Node) of
  161          pong ->
  162:             case rpc:call(Node,
  163                            application,
  164                            get_env,

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_capability.erl:
  606  query_capability(Node, Capability, DefaultSup, {App, Var, Map}) ->
  607      Default = {Capability, [DefaultSup]},
  608:     Result = rpc:call(Node, application, get_env, [App, Var]),
  609      case Result of
  610          {badrpc, _} ->

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_claim_sim.erl:
  120                  {Ring0, TN0};
  121              {Claimant, undefined, undefined} ->
  122:                 {ok, Ring0} = rpc:call(Claimant, riak_core_ring_manager, get_raw_ring, []),
  123:                 ClaimantTN = rpc:call(Claimant, app_helper, get_env, [riak_core, target_n_val]),
  124                  {Ring0, ClaimantTN};
  125              {undefined, RingFile, undefined} ->

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_console.erl:
  812              error;
  813          true ->
  814:             case rpc:call(Node, riak_core_handoff_manager,
  815                            set_concurrency, [Limit]) of
  816                  {badrpc, _} ->

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_gen_server.erl:
  625       rec_nodes_rest(Tag, Tail, Name, [N | Badnodes], Replies)
  626      after Time ->
  627:      case rpc:call(N, erlang, whereis, [Name]) of
  628       Pid when is_pid(Pid) -> % It exists try again.
  629           rec_nodes(Tag, [N|Tail], Name, Badnodes,

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_gossip.erl:
  194      case GossipVsn of
  195          undefined ->
  196:             case rpc:call(Node, riak_core_gossip, gossip_version, [], 1000) of
  197                  {badrpc, _} ->
  198                      ?LEGACY_RING_VSN;

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_gossip_legacy.erl:
  174      % Set the remote node to stop claiming.
  175      % Ignore return of rpc as this should succeed even if node is offline
  176:     rpc:call(ExitingNode, application, set_env,
  177               [riak_core, wants_claim_fun, {riak_core_claim, never_wants_claim}]),
  178  
  ...
  212      % Set the new ring on the exiting node. This will trigger
  213      % it to begin handoff and cleanly leave the cluster.
  214:     rpc:call(ExitingNode, riak_core_ring_manager, set_my_ring, [ExitRing]),
  215  
  216      ok.

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_handoff_sender.erl:
  430  
  431  get_handoff_ip(Node) when is_atom(Node) ->
  432:     case rpc:call(Node, riak_core_handoff_listener, get_handoff_ip, [],
  433                    infinity) of
  434          {badrpc, _} ->
  ...
  564  remote_supports_batching(Node) ->
  565  
  566:     case catch rpc:call(Node, riak_core_handoff_receiver, 
  567                    supports_batching, []) of
  568          true ->

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_net_ticktime.erl:
   42      EbinDir = filename:dirname(code:which(?MODULE)),
   43      try
   44:         Dirs = rpc:call(Node, code, get_path, []),
   45          case lists:member(EbinDir, Dirs) of
   46              false ->
   47                  lager:info("start_set_net_ticktime_daemon: adding to code path "
   48                             "for node ~p\n", [Node]),
   49:                 rpc:call(Node, code, add_pathz, [EbinDir]);
   50              true ->
   51                  ok
   ..
   81  stop_set_net_ticktime_daemon(Node, true) ->
   82      try
   83:         case rpc:call(Node, erlang, whereis, [?REGNAME]) of
   84              Pid when is_pid(Pid) ->
   85                  io:format("Stopping tick daemon ~p on ~p\n", [Pid, Node]),

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_status.erl:
   87      %% Check if the claimant is running and if it believes the ring is ready
   88      Claimant = riak_core_ring:claimant(Ring),
   89:     case rpc:call(Claimant, riak_core_ring, ring_ready, [], 5000) of
   90          {badrpc, _} ->
   91              Down2 = lists:usort([Claimant|Down]),

/Users/joe/dev/basho/riak_ee/deps/riak_core/src/riak_core_util.erl:
  220  reload_all(Module) ->
  221      {ok, Ring} = riak_core_ring_manager:get_my_ring(),
  222:     [{rpc:call(Node, code, purge, [Module]),
  223:      rpc:call(Node, code, load_file, [Module])} ||
  224          Node <- riak_core_ring:all_members(Ring)].
  225  
  ...
  419  multi_rpc(Nodes, Mod, Fun, Args, Timeout) ->
  420      pmap(fun(Node) ->
  421:                  rpc:call(Node, Mod, Fun, Args, Timeout)
  422           end, Nodes).
  423  
```

**Comment by joedevivo** [comment]
More details available: https://gist.github.com/reiddraper/7955381dc1fecf552220

**Comment by cmeiklejohn** [comment]
Closed by #586.
